sherpa/plot/bokeh_backend.py

```diff
@@ -0,0 +1,829 @@
+
+        # Axis sharing
+        for c in range(ncols):
+            basenum = None
+            for r in range(nrows):
+                index = self._index_axis(0, c)
+                if basenum is None:
+                    basenum = index
+                else:
+                    self.current_fig[index].x_range = \
+                        self.current_fig[basenum].x_range
+
+        plotnum = self._index_axis(row, col)
+        if plotnum is None:
+            raise ArgumentErr("No plot at row {} and column {}." +
+                              "Use create=True to create an entirely new, " +
+                              "complete grid of plots.".format(row, col)) from None
+        self.current_axis = self.current_fig[plotnum]
+
+
+    # HTML representation
+
+    def as_html(self, func):
+        """Create HTML representation of a plot
+
+
+        Parameters
+        ----------
+        func : function
+            The function, which takes no arguments, which will create the
+            plot. It creates the Figure.
+
+        Returns
+        -------
+        plot : str or None
+            The HTML, or None if there was an error (e.g. prepare not
+            called).
+
+        """
+        try:
+            func()
+        except Exception as e:
+            logger.debug("Unable to create bokeh plot: %s", str(e))
+            return None
+
+        image = embed.file_html(self.current_fig)
+
+        # See https://docs.bokeh.org/en/latest/docs/reference/embed.html#bokeh.embed.components
+        # for a list of other js files that might be needed.
+        # Also, do we want to use this offline instead of CDN?
+        return f'''
+        <script src="https://cdn.bokeh.org/bokeh/release/bokeh-{bokeh.__version__}.min.js"></script>
+        ''' + image
+
+
+    def as_html_plot_or_contour(self, data, summary=None, func="plot"):
+        """Create HTML representation of a plot
+
+        Parameters
+        ----------
+        data : Plot instance
+            The plot object to display. It has already had its prepare
+            method called.
+        summary : str or None, optional
+            The summary of the detail. If not set then the data type
+            name is used.
+        func : str, optional
+            The function to call on the data object to make the plot.
+
+        Returns
+        -------
+        plot : str or None
+            The HTML, or None if there was an error (e.g. prepare not
+            called).
+        """
+        image = self.as_html(getattr(data, func))
+
+        if image is None:
+            return None
+
+        if summary is None:
+            summary = type(data).__name__
+
+        ls = [formatting.html_svg(image, summary)]
+        return formatting.html_from_sections(data, ls)
+
+    as_html_plot = partialmethod(as_html_plot_or_contour, func="plot")
+    as_html_contour = partialmethod(as_html_plot_or_contour, func="contour")
```

Comment:
pylint is complaining to me about both lines (ie same reason for both); there's two items but I'm sure they are essentially  the same issue

- W4903: Using deprecated argument func of method partialmethod() (deprecated-argument)
- E1124: Argument 'func' passed by position and keyword in constructor call (redundant-keyword-arg)

I can't see any recent documentation to say when this behaviour changed

Reply:
I think pylist is misunderstanding here. `func` is the keyword name in `as_html_plot_or_contour`, not an argument to `partialmethod`. Maybe it was a long time ago, but the fact that this works shows that Python interprets this way way I intended. I can rename the argument in `as_html_plot_or_contour` though, just to make pylint happy.